src/lib/pipeline/m2/index.js

```diff
@@ -20,23 +21,31 @@ class M2 extends THREE.Group {
 
     const bones = [];
     const rootBones = [];
+    const indexedBones = [];
 
-    this.data.bones.forEach((joint) => {
+    this.data.bones.forEach((joint, index) => {
       const bone = new THREE.Bone();
 
-      const { pivotPoint } = joint;
+      indexedBones[index] = bone;
```

Comment:
I think you should be able to use `bones` as an index here. Or at least, I seem to have done the same thing a bit below:

```javascript
const parent = bones[joint.parentID];
```